integration-tests/cypress/commands/api/device.ts

```diff
@@ -0,0 +1,539 @@
+  		  expect(users[count].DeviceUsers.UserId).to.equal(expectedUsers[count].DeviceUsers.UserId);
+	      };
+  	    };
+    	  }
+       };
+  });
+});
+
+Cypress.Commands.add("apiCheckDevices_contains", (userName: string, expectedDevices: [ComparableDevice], params: string = {}, statusCode: number = 200) => {
+  const fullUrl = v1ApiPath('devices',params);
+  logTestDescription(
+      `${userName} Check devices seen by user '${userName}' contain the expected devices `,
+      { user: userName  },
+      true
+  );
+
+  makeAuthorizedRequestWithStatus(
+      {
+        method: "GET",
+        url: fullUrl,
+        body: null
+      },
+      userName,
+      statusCode
+    ).then((response)=>{
+       if(statusCode==null || statusCode==200) {
+          var devices=response.body.devices.rows;
+          expect(response.body.devices.count).to.be.at.least(expectedDevices.length);
+          expect(devices.length).to.be.at.least(expectedDevices.length);
+          var dev_count;
+          var user_count;
+	  //check each device in our expected list
+          for (dev_count=0; dev_count < expectedDevices.length; dev_count++) {    
+            var found=false;
+	    // is found somewhere i the actual list
+	    devices.forEach(function(device) {
+	      // and contains the correct values
+              if(device.devicename==expectedDevices[dev_count].devicename) {
+		found=true;
+		expect(device.id).to.equal(expectedDevices[dev_count].id);
+                expect(device.devicename).to.equal(expectedDevices[dev_count].devicename);
+                expect(device.active).to.equal(expectedDevices[dev_count].active);
+                if(expectedDevices[dev_count].Users==null) {
+                  expect(device.Users).to.not.exist;
+                } else {
+                  expect(device.Users.length).to.equal(expectedDevices[dev_count].Users.length);
+	          // sort users and expected users to ensure order is the same
+		  var users=sortArrayOn(device.Users,'username');
+		  var expectedUsers=sortArrayOn(expectedDevices[dev_count].Users,'username');
+
+	          // compare user list
+                  var count;
+                  for (count=0; count < expectedDevices[dev_count].Users.length; count++) {
+                      expect(users[count].username).to.equal(expectedUsers[count].username);
+                      expect(users[count].id).to.equal(expectedUsers[count].id);
+                      expect(users[count].DeviceUsers.admin).to.equal(expectedUsers[count].DeviceUsers.admin);
+                      expect(users[count].DeviceUsers.UserId).to.equal(expectedUsers[count].DeviceUsers.UserId);
+                  };
+                };
+              };
+	    });
+            expect(found).to.equal(true);
+          }
+       };
+  });
+});
+
+Cypress.Commands.add("apiCheckDeviceInGroup", (userName: string, cameraName: string, groupName: string, groupId: number, expectedDevice: all, params:string  = {}, statusCode: number = 200) => {
```

Comment:
if gorupId can be null should be spelled out here for tyescript

Reply:
Please clarify what is required

Reply:
Done - optional param in definitions file